### ISSUE TITLE
The object shortcut key of a nvdaControls.CustomCheckListBox is not spoken by NVDA

### ISSUE THREAD

**Issue by paulber19** [nvaccess/nvda#15816]
### Actual behavior:

In nvda settings, Check the "Report object shortcut keys" option of "Object Presentation" category.
- go to NVDA settings and "Keyboard" category.
- press "tab" twice.
- NVDA says:
"Select NVDA Modifier Keys list"
"caps lock check box checked"
The sshortcut key "alt+s" is not spoken.
### Expected behavior:

NVDA should say:
"Select NVDA Modifier Keys list alt+s"
"caps lock check box checked"
#### NVDA installed/portable/running from source:
Installed  and portable
#### NVDA version:
NVDA 2023.1, 2023.2 2023.3
#### Windows version:
Windows 10 22H2 (10.0.19045) workstation AMD64

#### Name and version of other software in use when reproducing the issue:
None
#### Does the issue still occur after restarting your computer?
Yes
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by CyrilleB79** [comment]
To know the shortcut of a list, you can use the following workaround:
When a list item is focused, object navigate to its parent pressing NVDA+pavNum8; this way, when the navigator object is on the whole list, the shortcut key is reported.

Note that the issue does only apply to `CustomCheckListBox` but also to other lists, at list in NVDA GUI:
* List of categories in the settings dialog
* List of symbols in the Punctuation/symbol pronunciation dialog.
* List of add-ons in the Add-on Store

The case of the dictionary entries list is a little more strange:
The list is first focused globally when the dialog is opened and when you tab / shift+tab in it. However, once you focus one item in the list, e.g. using arrows, you cannot focus anymore the whole list again.

To summarize, in the case of lists, the shortcut is not reported because the list item is focused, not the whole list.

Usually we do not report the shortcut key of a container when it is entered. I think that it would make sense to do an exception for lists.

**Comment by CyrilleB79** [comment]
To know the shortcut of a list, you can use the following workaround:
When a list item is focused, object navigate to its parent pressing NVDA+pavNum8; this way, when the navigator object is on the whole list, the shortcut key is reported.

Note that the issue does only apply to `CustomCheckListBox` but also to other lists, at list in NVDA GUI:
* List of categories in the settings dialog
* List of symbols in the Punctuation/symbol pronunciation dialog.
* List of add-ons in the Add-on Store

The case of the dictionary entries list is a little more strange:
The list is first focused globally when the dialog is opened and when you tab / shift+tab in it. However, once you focus one item in the list, e.g. using arrows, you cannot focus anymore the whole list again.

To summarize, in the case of lists, the shortcut is not reported because the list item is focused, not the whole list.

Usually we do not report the shortcut key of a container when it is entered. I think that it would make sense to do an exception for lists.

**Comment by amirmahdifard** [comment]
@CyrilleB79 also, the list in the configuration dialog doesn't have any lable: it just says list: and then the item in the list: it should have a lable such as profiles list

**Comment by CyrilleB79** [comment]
> @CyrilleB79 also, the list in the configuration dialog doesn't have any lable: it just says list: and then the item in the list: it should have a lable such as profiles list

That's true but that's a separate issue with respect to the current one; please open a new issue if it is important for you; note that there may be other unlabeled lists in the GUI; if you open an issue for this, a list of the unlabeled lists would be helpful.
